### ISSUE TITLE
WP Debug Log PHP Notice

### ISSUE THREAD

**Issue by carllockett3** [WordPress/gutenberg#10769]
**Describe the bug**
There is a PHP notice filling up the WP Debug log that I started noticing a couple of weeks ago. The 2 error lines keep repeating over and over. Anyway I just thought I would bring it to your attention. I searched for this on Github, but couldn't find anything concerning this. Here are the lines from the debug log:
```
[19-Oct-2018 01:24:59 UTC] PHP Notice: Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264 
[19-Oct-2018 01:24:59 UTC] PHP Notice: Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
```

I really don't know how to reproduce the errors. They seem to appear when I am creating a post. Each time it shows in the log there are at least 30 lines of the above two errors. within a span of about 1-2 seconds. I think it is happening during the auto saves.

**Expected behavior**
No error.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version 69.0.3497.100
 - Gutenberg version: 4.0.0
 - WordPress version: 4.9.8

This is a sample of the debug log:
```
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 14:56:27 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:00:15 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 262
[18-Oct-2018 15:01:07 UTC] PHP Notice:  Trying to get property of non-object in /home1/cilcreat/public_html/wp-content/plugins/gutenberg/gutenberg.php on line 264
```

**Comment by danielbachhuber** [comment]
Hi @carllockett3,

It seems that something might be stomping on the `$post` object here:

https://github.com/WordPress/gutenberg/blob/7936eea1747f42ebeff1f2b40f06eb149840bf22/gutenberg.php#L264

Can you try deactivating your plugins one by one to see if one of them might be breaking the use of the `post_row_actions` filter?

**Comment by carllockett3** [comment]
I solved this issue!
Sorry about the delay in getting back with you on this. I've spent almost all day trying to recreate this issue with no result. I did discover it's **my plugin** and now I feel really stupid! 

The issue happens with my WooCommerce products list which I am using an extension of WP_List_Table(). I had the following line in my code which was conflicting with Gutenberg. I had thought I installed Gutenberg on my test installation, but of course I hadn't! As soon as I installed Gutenberg, PHP very kindly showed the errors all over my screen! Sorry about this, but thankfully I found this issue before you rolled out Gutenberg with WP 5.0. 

Here is the line of code in question:
`$actions = apply_filters('page_row_actions', $actions, $item['id']);`

I really don't know why I had that line of code. The function works perfectly without it. I think it was in the original WP_LIST_TABLE() boilerplate template I used.
